test/common/common/logger_test.cc

```diff
@@ -0,0 +1,47 @@
+#include <string>
+
+#include "common/common/logger.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+
+class LoggerEscapeTest : public testing::Test {};
+
+TEST_F(LoggerEscapeTest, LinuxEOL) {
+  const std::string log = Logger::DelegatingLogSink::escapeLogLine("line 1 \n line 2\n");
```

Comment:
these tests are great. They give me a lot more confidence that your escaper covers the corner cases I had in mind.

One more case to add: an empty string.

Reply:
Done